Unit tests failing on Linux, mlx is macOS only

I'm no longer able to run the unit test suite successfully on my local laptop with a fresh tox virtual env. Several unit tests in `tests/test_lab_train.py` are failing. The attribute error from mock patch is a bit of a red herring. The tests are failing, because a fresh virtual env does not have the `mlx` package installed. The `mlx` package is Apple Silicon-only.

Our CI is currently not failing due to a caching bug in our GHA spec. We have been using and updating an old cached `.tox` directory that happens to have mlx 0.2.0 installed. It's the most recent version that has Linux binaries. The project no longer releases sdist and Linux binaries. Our CI will fail as soon as we loose access to the cache. Also see #673 and #675.

The new tests were introduced in #684 (CC @danmcp). I guess Dan is on macOS and therefore never noticed the problem in the first place.

```
============================================== short test summary info ==============================================
FAILED tests/test_lab_train.py::TestLabTrain::test_invalid_data_dir_synthetic - AssertionError: Expected 'make_data' to have been called once. Called 0 times.
FAILED tests/test_lab_train.py::TestLabTrain::test_load - AttributeError: module 'cli.train.lora_mlx' has no attribute 'lora'
FAILED tests/test_lab_train.py::TestLabTrain::test_load_local - AttributeError: module 'cli.train.lora_mlx' has no attribute 'lora'
FAILED tests/test_lab_train.py::TestLabTrain::test_skip_preprocessing - AttributeError: module 'cli.train.lora_mlx' has no attribute 'lora'
FAILED tests/test_lab_train.py::TestLabTrain::test_skip_quantize - AttributeError: module 'cli.train.lora_mlx' has no attribute 'lora'
FAILED tests/test_lab_train.py::TestLabTrain::test_train_mac - AttributeError: module 'cli.train.lora_mlx' has no attribute 'lora'
===================================== 6 failed, 27 passed, 3 warnings in 6.48s ======================================
```

--- a/tests/test_lab_train.py
+++ b/tests/test_lab_train.py
@@ -1,5 +1,5 @@
 # Standard
-from unittest.mock import patch
+from unittest.mock import MagicMock, patch
 import os
 import platform
 import sys
@@ -22,6 +22,13 @@ FINAL_RESULTS_DIR = TRAINING_RESULTS_DIR + "/" + FINAL_RESULTS_DIR_NAME
 LINUX_GGUF_FILE = FINAL_RESULTS_DIR + "/ggml-model-f16.gguf"
 MODEL_DIR = "model"
 ENCODING = "UTF-8"
+
+# mlx isn't imported for Linux
+sys.modules["mlx"] = MagicMock()
+sys.modules["mlx.utils"] = MagicMock()
+sys.modules["mlx.core"] = MagicMock()
+sys.modules["mlx.nn"] = MagicMock()
+sys.modules["mlx.optimizers"] = MagicMock()
 
 
 def setup_input_dir():

